code/modules/religion/festival/instrument_rites.dm

```diff
@@ -0,0 +1,149 @@
+			continue
+		if(!listener.ckey) //good requirement to have for favor, trust me
+			continue
+		GLOB.religious_sect.adjust_favor(0.2)
+
+/datum/religion_rites/song_tuner/evangelism/finish_effect(atom/song_player, datum/song/song_datum)
+	for(var/mob/living/carbon/human/listener in song_datum.hearing_mobs)
+		SEND_SIGNAL(listener, COMSIG_ADD_MOOD_EVENT, "blessing", /datum/mood_event/blessing)
+
+/datum/religion_rites/song_tuner/nullwave
+	name = "Nullwave Vibrato"
+	desc = "Sing a dull song, protecting those who listen from magic."
+	particles_path = /particles/musical_notes/nullwave
+	song_invocation_message = "You've prepared an antimagic song!"
+	song_start_message = span_nicegreen("This music makes you feel protected!")
+	glow_color = "#a9a9b8"
+	repeats_okay = FALSE
+
+/datum/religion_rites/song_tuner/nullwave/song_effect(atom/song_player, datum/song/song_datum)
+	if(!song_datum)
+		return
+	for(var/mob/living/listener in song_datum.hearing_mobs)
+		if(listener.anti_magic_check(magic = FALSE, holy = TRUE))
+			continue
+		listener.apply_status_effect(/datum/status_effect/antimagic, 2 MINUTES)
+
+/datum/religion_rites/song_tuner/pain
+	name = "Murderous Chord"
+	desc = "Sing a sharp song, cutting those around you. Works less effectively on fellow priests. At the end of the song, you'll open the wounds of all listeners."
+	particles_path = /particles/musical_notes/harm
+	song_invocation_message = "You've prepared a painful song!"
+	song_start_message = span_danger("This music cuts like a knife!")
+	glow_color = "#FF4460"
+	repeats_okay = FALSE
+
+/datum/religion_rites/song_tuner/pain/song_effect(atom/song_player, datum/song/song_datum)
+	if(!song_datum)
+		return
+	for(var/mob/living/listener in song_datum.hearing_mobs)
+		if(listener.anti_magic_check(magic = FALSE, holy = TRUE))
+			continue
+		var/pain_juice = 1
+		if(listener.mind?.holy_role)
+			pain_juice *= 0.5
+		listener.adjustBruteLoss(pain_juice)
+
+/datum/religion_rites/song_tuner/pain/finish_effect(atom/song_player, datum/song/song_datum)
+	for(var/mob/living/carbon/human/listener in song_datum.hearing_mobs)
+		if(listener.anti_magic_check(magic = FALSE, holy = TRUE))
+			continue
+		var/obj/item/bodypart/sliced_limb = pick(listener.bodyparts)
+		sliced_limb.force_wound_upwards(/datum/wound/slash/moderate/many_cuts)
+
+/datum/religion_rites/song_tuner/lullaby
+	name = "Spiritual Lullaby"
+	desc = "Sing a lullaby, tiring those around you, making them slower. At the end of the song, you'll put people who are tired enough to sleep."
+	particles_path = /particles/musical_notes/sleepy
+	song_invocation_message = "You've prepared a sleepy song!"
+	song_start_message = span_warning("This music's making you feel drowsy...")
+	favor_cost = 40 //actually really strong
+	glow_color = "#83F6FF"
+	repeats_okay = FALSE
+	///assoc list of weakrefs to who heard the song and
+	var/list/listener_counter = list()
+
+/datum/religion_rites/song_tuner/lullaby/Destroy()
+	QDEL_NULL(listener_counter)
```

Comment:
`listener_count` is a list and you're just qdeling it directly, it should probably be a `Cut()` or something
```suggestion
	listener_counter.Cut()
```